src/base/net/downloadhandlerimpl.cpp

```diff
@@ -121,9 +124,13 @@ void DownloadHandlerImpl::processFinishedDownload()
     }
 
     // Success
+#if (QT_VERSION >= QT_VERSION_CHECK(6, 3, 0))
+    m_result.data = m_reply->readAll();
+#else
     m_result.data = (m_reply->rawHeader("Content-Encoding") == "gzip")
                     ? Utils::Gzip::decompress(m_reply->readAll())
                     : m_reply->readAll();
+#endif
```

Comment:
```suggestion
    QByteArray rawData = m_reply->readAll();
    m_result.data = (rawData.at(1) == 0x1f && rawData.at(2) == 0x8b)
                    ? Utils::Gzip::decompress(rawData)
                    : rawData;
```
@glassez 
Find some reference from here https://stackoverflow.com/questions/6059302/how-to-check-if-a-file-is-gzip-compressed.

Test with Qt 6.3 and it works. But I don't know whether it will work for Qt < 6.3. Besides, I think this method is somewhat hacky.

Reply:
> Besides, I think this method is somewhat hacky.

Is this some kind of documented watermark of the Gzip format that you use in your suggestion above? If Yes, I wouldn't call this method hacky.

Reply:
I think compressed file and compress method should be known from `content-encoding` header.

Because Qt also support `Brotli` (known as `content-encoding: br` header): https://doc.qt.io/qt-5/qtwebengine-3rdparty-brotli.html.

Maybe we should also handle Brotli ?

Reply:
> Is this some kind of documented watermark of the Gzip format that you use in your suggestion above?

From here: https://www.ietf.org/rfc/rfc1952.txt
```

      2.3.1. Member header and trailer

         ID1 (IDentification 1)
         ID2 (IDentification 2)
            These have the fixed values ID1 = 31 (0x1f, \037), ID2 = 139
            (0x8b, \213), to identify the file as being in gzip format.

         CM (Compression Method)
            This identifies the compression method used in the file.  CM
            = 0-7 are reserved.  CM = 8 denotes the "deflate"
            compression method, which is the one customarily used by
            gzip and which is documented elsewhere.
```



> Maybe we should also handle Brotli ?

https://github.com/qbittorrent/qBittorrent/blob/fd5136254539d601b1b0d03de28bf70b8a65583c/src/base/net/downloadmanager.cpp#L127

___________________________

@glassez @Chocobo1 

Do we really need to address this situation ourself instead of relying on upstream Qt? I'm open to this because of we only accept the GZip encoding after all.

Reply:
>Do we really need to address this situation ourself instead of relying on upstream Qt? 

I would prefer to rely on Qt. However it is strange that Qt 6.3 made a (big?) breaking change and it isn't mentioned on their change log.

>Besides, I think this method is somewhat hacky.

I thought so too, detection at this location could have false positives AFAIK.

Reply:
> However it is strange that Qt 6.3 made a (big?) breaking change and it isn't mentioned on their change log.

That's exactly what I'm worried about. But let's leave it as it is for now and see what happens next.